test/test_lookout.py

```diff
@@ -709,6 +709,49 @@ def test_failover_master_one_standby_one_observer_no_connections(pgl):
     assert pgl.execute_external_command.call_count == 1
 
 
+def test_failover_master_one_standby_no_observer_no_connections(pgl):
+    pgl.own_db = "this_host"
+    pgl.current_master = "primary"
+
+    # add db state
+    _add_db_to_cluster_state(pgl, "primary", pg_is_in_recovery=False, connection=False)
+    _add_db_to_cluster_state(
+        pgl,
+        "this_host",
+        pg_last_xlog_receive_location="2/aaaaaaaa",
+        pg_is_in_recovery=True,
+        connection=True,
+        replication_time_lag=1.0,
+    )
+
+    # primary is still considered to be part of the cluster even if we're disconnected
+    pgl.check_cluster_state()
+    assert pgl.execute_external_command.call_count == 0
+
+    # age the configuration, we should still be fine as we've seen the primary at `utcnow`
+    pgl.cluster_nodes_change_time = time.monotonic() - pgl.missing_master_from_config_timeout - 1
+    _add_db_to_cluster_state(
```

Comment:
As we're clearly using this function `_add_db_to_cluster_state` not just to add but also to update / set, what do you think of updating its name? Maybe `_set_db_to_cluster_state`?

Reply:
Yeah, this needs to be changed but in a separate PR. I did stumble over it while making the changes. I'll add a ticket to change it.

Reply:
Thanks!